test/krew/krew.go

```diff
@@ -32,51 +33,82 @@ const krewBinaryEnv = "KREW_BINARY"
 
 // ITest is used to set up `krew` integration tests.
 type ITest struct {
-	t       *testing.T
-	plugin  string
-	args    []string
-	env     []string
-	tempDir *testutil.TempDir
+	t             *testing.T
+	plugin        string
+	pluginsBinDir string
+	args          []string
+	env           []string
+	tempDir       *testutil.TempDir
 }
 
 // NewTest creates a fluent krew ITest.
 func NewTest(t *testing.T) (*ITest, func()) {
 	tempDir, cleanup := testutil.NewTempDir(t)
-	pathEnv := setupPathEnv(t, tempDir)
+	binDir := setupKrewBin(t, tempDir)
+
 	return &ITest{
-		t:       t,
-		env:     []string{pathEnv, fmt.Sprintf("KREW_ROOT=%s", tempDir.Root())},
+		t:             t,
+		pluginsBinDir: binDir,
+		env: []string{
+			fmt.Sprintf("KREW_ROOT=%s", tempDir.Root()),
+			fmt.Sprintf("PATH=%s", augmentPATH(t, binDir)),
+		},
 		tempDir: tempDir,
 	}, cleanup
 }
 
-func setupPathEnv(t *testing.T, tempDir *testutil.TempDir) string {
-	krewBinPath := tempDir.Path("bin")
-	if err := os.MkdirAll(krewBinPath, os.ModePerm); err != nil {
+// setupKrewBin symlinks the $KREW_BINARY to $tempDir/bin and returns the path
+// to this directory.
+func setupKrewBin(t *testing.T, tempDir *testutil.TempDir) string {
+	krewBinary, found := os.LookupEnv(krewBinaryEnv)
+	if !found {
+		t.Fatalf("%s environment variable pointing to krew binary not set", krewBinaryEnv)
+	}
+	binPath := tempDir.Path("bin")
+	if err := os.MkdirAll(binPath, 0755); err != nil {
```

Comment:
This `mkdir` functionality could also be moved to `TempDir`. Will avoid further repetition in the future.

Reply:
I think it's ok, we might introduce too much magic if we move. It's not too repetitive for now.